manifest-generator/generator-services/src/main/java/io/hyscale/generator/services/plugins/NetworkPoliciesHandler.java

```
@@ -0,0 +1,90 @@
+import org.springframework.stereotype.Component;
+import java.util.HashMap;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+
+@Component
+@ManifestPlugin(name = "NetworkPoliciesHandler")
+public class NetworkPoliciesHandler implements ManifestHandler {
+
+    @Autowired
+    private MustacheTemplateResolver templateResolver;
+
+    @Autowired
+    private PluginTemplateProvider templateProvider;
+
+    private static final Logger logger = LoggerFactory.getLogger(NetworkPoliciesHandler.class);
+    private static final String SERVICE_NAME = "service_name";
+    private static final String NETWORK_RULES = "rules";
+
+    @Override
+    public List<ManifestSnippet> handle(ServiceSpec serviceSpec, ManifestContext manifestContext) throws HyscaleException {
+        // Check if the Spec is External
+        if (serviceSpec.get(HyscaleSpecFields.external) == null || serviceSpec.get(HyscaleSpecFields.external, Boolean.class)) {
+            return null;
+        }
+        try {
+            // All Network Traffic Rules Go Under Allow Traffic
+            List<NetworkTrafficRule> networkTrafficRules = serviceSpec.get(HyscaleSpecFields.allowTraffic, new TypeReference<>() {
+            });
+
+            // To Generate k8s YAML Template
+            ConfigTemplate networkPoliciesTemplate = templateProvider.get(PluginTemplateProvider.PluginTemplateType.NETWORK_POLICY);
+            String yamlString = templateResolver.resolveTemplate(networkPoliciesTemplate.getTemplatePath(), getContext(networkTrafficRules, serviceSpec, manifestContext));
+
+            ManifestSnippet snippet = new ManifestSnippet();
+            snippet.setKind(ManifestResource.NETWORK_POLICY.getKind());
+            snippet.setPath("spec");
+            snippet.setSnippet(yamlString);
+
+            List<ManifestSnippet> snippetList = new LinkedList<>();
+            snippetList.add(snippet);
+            return snippetList;
+        } catch (Exception e) {
```

Rethrow the exception as Hyscale Exception instead of returning null.